beerus_core/src/lightclient/starknet/mod.rs

```diff
@@ -22,4 +32,38 @@ impl StarkNetLightClient for StarkNetLightClientImpl {
     async fn start(&mut self) -> Result<()> {
         Ok(())
     }
+
+    /// Call a contract on StarkNet.
+    /// Returns the result of the call.
+    ///
+    /// # Arguments
+    ///
+    /// * `contract_address` - Address of the contract.
+    /// * `selector` - Selector of the function to call.
+    /// * `calldata` - Calldata of the function to call.
+    ///
+    /// # Returns
+    ///
+    /// `Ok(Vec<FieldElement>)` if the operation was successful.
+    /// `Err(eyre::Report)` if the operation failed.
+    pub async fn call(
+        &self,
+        contract_address: FieldElement,
+        entry_point_selector: FieldElement,
+        calldata: Vec<FieldElement>,
+    ) -> Result<Vec<FieldElement>> {
+        let request: FunctionCall = FunctionCall {
+            contract_address,
+            entry_point_selector,
+            calldata,
+        };
+        let response = self
+            .client
+            .call(
+                request,
+                &starknet::providers::jsonrpc::models::BlockId::Number(485441),
```

Comment:
block number is hardcoded, use latest block retrieved from StarkNet core contract